connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,258 @@
+           return readParquetFile(parquetFile, parquetFileStream);
+        } catch (IOException e) {
+            throw new TranslatorException(e);
+        }
+    }
+
+    private RecordReader<Group> readParquetFile(VirtualFile parquetFile, InputStream parquetFileStream) throws TranslatorException {
+       try {
+           File localFile = createTempFile(parquetFileStream);
+           Path path = new Path(localFile.toURI());
+           Configuration config = new Configuration();
+           reader = ParquetFileReader.open(HadoopInputFile.fromPath(path, config));
+           schema = reader.getFooter().getFileMetaData().getSchema();
+           MessageType filteredSchema = getFilteredSchema(schema, this.visitor.getProjectedColumnNames());
+           columnIO = new ColumnIOFactory().getColumnIO(filteredSchema);
+           PageReadStore pages = reader.readNextRowGroup();
+           pageRowCount = pages.getRowCount();
+           return columnIO.getRecordReader(pages, new GroupRecordConverter(filteredSchema));
+       } catch (IOException e){
+           throw new TranslatorException(e);
+       }
+    }
+
+    private MessageType getFilteredSchema(MessageType schema, List<String> expectedColumnNames) {
+        List<Type> allColumns = schema.getFields();
+        List<Type> selectedColumns = new ArrayList<>();
+        for(String columnName: expectedColumnNames) {
+            selectedColumns.add(allColumns.get(schema.getFieldIndex(columnName)));
+        }
+        MessageType filteredSchema = new MessageType(schema.getName(), selectedColumns);
+        return filteredSchema;
+    }
+
+
+    private String getExtension(String name) {
+        String extension = "";
+        int i = name.lastIndexOf('.');
+        if (i > 0) {
+            extension = name.substring(i+1);
+        }
+        return extension;
+    }
+
+    private File createTempFile(InputStream parquetFileStream) throws IOException {
+        File tempFile = File.createTempFile(getClass().getSimpleName(), ".tmp");
+        tempFile.deleteOnExit();
+        try (FileOutputStream out = new FileOutputStream(tempFile)) {
+            IOUtils.copyBytes(parquetFileStream, out, 131072);
+        }
+        return tempFile;
+    }
+
+    public Group nextRow() throws TranslatorException, DataNotAvailableException {
+            try {
+                Group row = nextRowInternal();
+                if (row == null) {
+                    return null;
+                }
+                return row;
+            } catch (IOException e) {
+                throw new TranslatorException(e);
+            }
+    }
+
+    private Group nextRowInternal() throws IOException, TranslatorException {
+        while (rowIterator != null) {
+            if (pageRowCount-- <= 0) {
+                this.rowIterator = null;
+                PageReadStore nextRowGroup = this.reader.readNextRowGroup();
+                if(nextRowGroup == null){
```

Comment:
I didn't handle the case of nextRowGroup != null here.  You'll need to account for that by resetting the pageRowCount and the rowIterator.